lib/krikri/enricher.rb

```diff
@@ -0,0 +1,132 @@
+module Krikri
+  ##
+  # A SoftwareAgent that runs enrichment processes.
+  #
+  # @example
+  #
+  #   To enrich records that were mapped by the mapping activity with ID 3:
+  #
+  #   # Define which enrichments are run, and thier parameters:
+  #   chain = {
+  #     'Krikri::Enrichments::StripHtml' => {
+  #       input_fields: [{sourceResource: :title}]
+  #     },
+  #     'Krikri::Enrichments::StripWhitespace' => {
+  #       input_fields: [{sourceResource: :title}]
+  #     }
+  #   }
+  #   Krikri::Enricher.enqueue({
+  #     generator_uri: 'http://ldp.local.dp.la/ldp/activity/3',
+  #     chain: chain
+  #   })
+  #
+  # @see Krikri::SoftwareAgent#enqueue
+  #
+  class Enricher
+    include SoftwareAgent
+
+    attr_reader :chain, :generator_uri
+
+    def self.queue_name
+      :enrichment
+    end
+
+    def initialize(opts = {})
+      @generator_uri = RDF::URI(opts.fetch(:generator_uri))
+      @chain = deep_sym(opts.fetch(:chain) { {} })
+    end
+
+    def run(activity_uri = nil)
+      log :info, 'enricher is running'
+      # see TODO below
+      target_aggregations.each do |agg|
+        chain_enrichments!(agg)
+        agg << RDF::Statement(agg, RDF::PROV.wasGeneratedBy, activity_uri) if
```

Comment:
We need a different predicate, here.  See, http://www.w3.org/TR/prov-o/#wasGeneratedBy

> Generation is the completion of production of a new entity by an activity. This entity did not exist before generation and becomes available for usage after this generation.

I'm looking into what should replace this.